Microsoft.Azure.Cosmos/src/Linq/TranslationContext.cs

```
@@ -81,10 +81,25 @@ public TranslationContext(CosmosLinqSerializerOptions linqSerializerOptions, IDi
             this.collectionStack = new List<Collection>();
             this.CurrentQuery = new QueryUnderConstruction(this.GetGenFreshParameterFunc());
             this.subqueryBindingStack = new Stack<SubqueryBinding>();
-            this.LinqSerializerOptions = linqSerializerOptions;
+            this.LinqSerializerOptions = linqSerializerOptions ?? new CosmosLinqSerializerOptions();
             this.Parameters = parameters;
-            this.MemberNames = new MemberNames(linqSerializerOptions);
-            this.CosmosLinqSerializer = new DefaultCosmosLinqSerializer();
+
+            if (this.LinqSerializerOptions.LinqSerializerType == LinqSerializerType.CustomCosmosSerializer)
+            {
+                if (this.LinqSerializerOptions.CustomCosmosSerializer == null)
+                {
+                    //Todo mayapainter: do we want to throw error here if SerializeLinqMemberName not implemented or let it throw a notimplementedexception from SerializeLinqMemberName?
```

I think throwing here is the right behavior:
Fail early, fail fast.

---

Not very confident about how this check is implemented.  Please lmk if you know of a better way to verify that the method is overriden.